### PR TITLE
break and continue cannot be used outside of loop

### PR DESCRIPTION
**Pull request by stepancheg**
It is an error even if code is not executed (static error).

Spec: https://git.io/fjipM

### CODE REVIEW

File: starlark/src/syntax/ast.rs [hunk 1]
```diff
@@ -335,6 +337,60 @@ impl Statement {
...
+            Statement::Break | Statement::Continue => {
+                let kw = if let Statement::Break = stmt.node {
+                    "break"
+                } else {
+                    "continue"
+                };
+                Err(Diagnostic {
+                    level: Level::Error,
+                    message: format!("{} cannot be used outside of loop", kw),
+                    code: Some("CS08".to_owned()),
```
Comment (damienmg) [inline starlark/src/syntax/ast.rs]: Extract the error code like for the rest of the module.

BTW if you have an idea on how to organize those error codes, feel free to share it.

File: starlark/src/syntax/ast.rs [hunk 2]
```diff
@@ -335,6 +337,60 @@ impl Statement {
...
+            | Statement::Load(..) => {
+                // These statements do not contain nested statements
+                Ok(())
+            }
+        }
+    }
+
+    pub(crate) fn validate_mod(stmt: &AstStatement, _dialect: Dialect) -> Result<(), Diagnostic> {
+        Statement::validate_break_continue(stmt)?;
+        Ok(())
```
[the remark this answers is not in the document]
Comment (stepancheg) [reply starlark/src/syntax/ast.rs]: That way readers of code may be confused assuming that this function may alter the tree, why it only validates it. But not a big deal.

**Comment by damienmg** [comment]
Ok that's fair, feel free to keep it that way.

File: starlark/src/syntax/ast.rs (hunk 1 above)
Comment (stepancheg) [reply starlark/src/syntax/ast.rs]: > Extract the error code like for the rest of the module.

In the rest of the module error codes are defined inline, but it's good idea to make constants for them. I'll do it.

> if you have an idea on how to organize those error codes, feel free to share it.

I'd do it like this: make a single big module with error code. Allocate two prefixes: `CT` for compile-time errors (like parser errors or break outside of loop) and `RT` for runtime errors like index out of range.

**Pull request merged**